File: tests/machine/x/dart/slice.dart
void main() {
  print((([1, 2, 3] is String) ? [1, 2, 3].substring(1, 3) : ([1, 2, 3] as List).sublist(1, 3)));
  print((([1, 2, 3] is String) ? [1, 2, 3].substring(0, 2) : ([1, 2, 3] as List).sublist(0, 2)));
  print((('hello' is String) ? 'hello'.substring(1, 4) : ('hello' as List).sublist(1, 4)));
}

File: tests/machine/x/java/slice.java
import java.util.*;

public class Slice {
    public static void main(String[] args) {
    System.out.println(((List)Arrays.asList(1, 2, 3)).subList(1, 3));
    System.out.println(((List)Arrays.asList(1, 2, 3)).subList(0, 2));
    System.out.println("hello".substring(1, 4));
    }
}
